pytest/tests/sanity/sync_ban.py

```diff
@@ -19,97 +19,112 @@
 from proxy import ProxyHandler
 import utils
 
-should_ban = sys.argv[1] == 'true'
-
 TIMEOUT = 300
 EPOCH_LENGTH = 50
 
 BAN_STRING = 'ban a fraudulent peer'
 
-should_sync = Value('i', False)
-
 
 class Handler(ProxyHandler):
 
+    def __init__(self, *args, should_sync=None, should_ban=None, **kwargs):
+        assert should_sync is not None
+        self.should_sync = should_sync
+        assert should_ban is not None
+        self.should_ban = should_ban
+        super().__init__(*args, **kwargs)
+
     async def handle(self, msg, fr, to):
         if msg is not None:
             if msg.enum == 'Block':
                 loop = asyncio.get_running_loop()
                 send = functools.partial(self.do_send_message, msg, 1)
-                if should_sync.value:
+                if self.should_sync.value:
                     loop.call_later(1, send)
                 return False
             elif msg.enum == 'BlockRequest':
                 loop = asyncio.get_running_loop()
                 send = functools.partial(self.do_send_message, msg, 0)
-                if should_sync.value:
+                if self.should_sync.value:
                     loop.call_later(6, send)
                 return False
             elif msg.enum == 'BlockHeaders':
-                if should_ban:
+                if self.should_ban:
                     return False
                 loop = asyncio.get_running_loop()
                 send = functools.partial(self.do_send_message, msg, 1)
-                if should_sync.value:
+                if self.should_sync.value:
                     loop.call_later(2, send)
                 return False
         return True
 
 
-node0_config = {
-    "consensus": {
-        "min_block_production_delay": {
-            "secs": 0,
-            "nanos": 400000000
+if __name__ == '__main__':
+    should_ban = sys.argv[1] == 'true'
+    node0_config = {
+        "consensus": {
+            "min_block_production_delay": {
+                "secs": 0,
+                "nanos": 1000000000
```

Comment:
why this change?

Reply:
The original test wanted it produce blocks too fast that the syncer chouldn't catch up as quickly (idk why but I don't wanna debug this moment). So I slowed it down.